tools/ld-chroma-decoder/encoder/CMakeLists.txt

```diff
@@ -0,0 +1,67 @@
+message( STATUS "---------Beginning cmake build of ld-chroma-encoder---------")
+
+project( ld-chroma-encoder )
+cmake_minimum_required( VERSION 3.22 )
+
+#ld-chroma-encoder SOURCES
+set ( SOURCES
+	"main.cpp"
+	"encoder.cpp"
+	"ntscencoder.cpp"
+	"palencoder.cpp"
+	"../../library/tbc/dropouts.cpp"
+	"../../library/tbc/jsonio.cpp"
+	"../../library/tbc/lddecodemetadata.cpp"
+	"../../library/tbc/logging.cpp"
+	"../../library/tbc/vbidecoder.cpp"
+)
+
+#ld-chroma-encoder HEADERS
+set ( HEADERS
+	"encoder.h"
+	"ntscencoder.h"
+	"palencoder.h"
+	"../../library/filter/firfilter.h"
+	"../../library/tbc/dropouts.h"
+	"../../library/tbc/jsonio.h"
+	"../../library/tbc/lddecodemetadata.h"
+	"../../library/tbc/logging.h"
+	"../../library/tbc/vbidecoder.h"
+)
+
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+
+set(CMAKE_CXX_STANDARD 17)
+
+# Get the current working branch
+execute_process(
+	COMMAND git rev-parse --abbrev-ref HEAD
+	WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+	OUTPUT_VARIABLE GIT_BRANCH
+	OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+
+# Get the latest abbreviated commit hash of the working branch
+execute_process(
+	COMMAND git log -1 --format=%h
+	WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+	OUTPUT_VARIABLE GIT_COMMIT_HASH
+	OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+
+# Pass it on to the source code
+add_definitions(-DAPP_BRANCH="\""+${GIT_BRANCH}+"\"" -DAPP_COMMIT="\""+${GIT_COMMIT_HASH}+"\"")
+
+find_package(Qt5 REQUIRED COMPONENTS Core)
+
+ADD_DEFINITIONS(${QT_DEFINITIONS})
+
+QT_WRAP_CPP(HEADERS_MOC ${HEADERS})
+
+# ld-chroma-encoder executable
+
+add_executable(${PROJECT_NAME} ${SOURCES} ${HEADERS_MOC})
+
+target_link_libraries(${PROJECT_NAME} PRIVATE Qt5::Core fftw3 ld-tbc ld-filter)
```

Comment:
The only tools that need FFTW are ld-analyse and ld-chroma-decoder - it's used by the Transform PAL decoder.